sklearnex/linear_model/tests/test_linear.py

```diff
@@ -81,3 +82,46 @@ def test_sklearnex_import_elastic():
     assert "daal4py" in elasticnet.__module__
     assert_allclose(elasticnet.intercept_, 1.451, atol=1e-3)
     assert_allclose(elasticnet.coef_, [18.838, 64.559], atol=1e-3)
+
+
+@pytest.mark.skipif(
+    not is_dpctl_available(["gpu"]),
+    reason="gpu_blas_compute_mode is only available for select gpus",
+)
+def test_bf16_blas_epsilon():
+    from sklearnex import config_context
+    from sklearnex.linear_model import LinearRegression
+
+    X = np.array([[0.0, 1.0], [1.0, 1.0]], dtype=np.float32)
+    y = np.array([[1.0], [1.0 + np.finfo(np.float32).eps]], dtype=np.float32)
+
+    # Make changes in y not observable in bfloat16 which means casting
+    # to bf16 for gemm should yield an incorrect value, everything
+    # should be fine until the end multiplication (inverses etc.).
+
+    with config_context(
+        target_offload="gpu:0",
+        allow_fallback_to_host=False,
+        gpu_blas_compute_mode="standard",
+    ):
+        linreg_standard = LinearRegression().fit(X, y)
+
+    with config_context(
+        target_offload="gpu:0",
+        allow_fallback_to_host=False,
+        gpu_blas_compute_mode="float_to_bf16",
+    ):
+        linreg_bf16 = LinearRegression().fit(X, y)
+
+    assert linreg_standard.n_features_in_ == 2
+    assert linreg_bf16.n_features_in_ == 2
+    # DAL_BLAS_COMPUTE_MODE is available in the mkl_gpu_fpk in 2024.0.2
+    if daal_check_version((2024, "P", 100)):

```

Comment:
What about NOT testing it in previous versions at all?

Reply:
I was hoping to verify operation with and without bf16 enablement. This seemed to me to be the best way of having my cake and eating it too.

Reply:
Could you please clarify why do you need to test it for not bf16 enablement?